src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ItemsPresenter.cs

```diff
@@ -0,0 +1,741 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media;
+using FluentAssertions;
+using MUXControlsTestApp.Utilities;
+using static Private.Infrastructure.TestServices;
+using Uno.UI.RuntimeTests.Extensions;
+using Windows.UI.Xaml;
+using Uno.Extensions;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls;
+
+public static class AssertionExtensions
+{
+	public static AndConstraint<FluentAssertions.Collections.GenericCollectionAssertions<float>> BeEquivalentToWithTolerance<T>(this FluentAssertions.Collections.GenericCollectionAssertions<float> assertions, float[] expected, float tolerance)
+	{
+		assertions.Subject.Should().HaveCount(expected.Length);
+
+		assertions.Subject.Zip(expected, (sub, actual) => sub.Should().BeApproximately(actual, tolerance)).ForEach((AndConstraint<FluentAssertions.Numeric.NumericAssertions<float>> _) => { });
+		return new AndConstraint<FluentAssertions.Collections.GenericCollectionAssertions<float>>(assertions);
+	}
+}
+
+[TestClass]
+public class Given_ItemsPresenter
+{
+	private float Epsilon =>
+#if XAMARIN
+			0.5f;
+#else
+		0.0f;
+#endif
+
+
+	[TestMethod]
+	[RunsOnUIThread]
+	[RequiresFullWindow]
+	public async Task When_Big_Elements_Horizontal_Many_Margins()
+	{
+
+		var grid = (Grid)XamlReader.Load("""
```

Comment:
With raw literals, you should be able to have less indentation for that block. Same for the tests below.